### DISCUSSION TITLE
Indexer System and Index Provider

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/lotus#8341] [Storage Provider]
Coming soon.

**Comment by Shekelme** [comment]
And while we're all waiting, I'll duplicate my question here.
After studying [this howto](https://lotus.filecoin.io/storage-providers/operate/index-provider/#one-off-migration), it remains unclear whether a monolithic lotus-miner process can act as an index provider? It seems to be possible, but then it is not described how to initialize it in this case. I'm afraid of making some fatal mistake.)
In step 7 ("One-off migration"), the manual says about a separate launch of the markets process, which (obviously) cannot be done in the case of a monolithic lotus-miner process. That's why I'm asking the question.

**Reply by jennijuju** [reply]
@Shekelme good q - the answer is the index provider can be run with the market subsystem, no matter how your setup is (monolith or seperated)
@LexLuthr can you clarify that in the docs?

**Reply by LexLuthr** [reply]
https://github.com/filecoin-project/lotus-docs/pull/165 is in progress.

**Reply by Shekelme** [reply]
Yep, I have already upgraded 2 monolith miners...
